cpp/include/cudf/io/experimental/hybrid_scan.hpp

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/io/parquet.hpp>
+#include <cudf/io/parquet_schema.hpp>
+#include <cudf/io/text/byte_range_info.hpp>
+#include <cudf/io/types.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/export.hpp>
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+namespace CUDF_EXPORT cudf {
+namespace io::parquet::experimental::detail {
+/**
+ * @brief Internal experimental Parquet reader optimized for highly selective filters, called a
+ *        Hybrid Scan operation.
+ */
+class hybrid_scan_reader_impl;
+}  // namespace io::parquet::experimental::detail
+}  // namespace CUDF_EXPORT cudf
+
+//! Using `byte_range_info` from cudf::io::text
+using cudf::io::text::byte_range_info;
+
+namespace CUDF_EXPORT cudf {
+namespace io::parquet::experimental {
+/**
+ * @addtogroup io_readers
+ * @{
+ * @file
+ */
+
+/**
+ * @brief The experimental parquet reader class to optimally read parquet files subject to
+ *        highly selective filters, called a Hybrid Scan operation
+ *
+ * This class is designed to best exploit reductive optimization techniques to speed up reading
+ * Parquet files subject to highly selective filters. This class reads file contents in two passes,
+ * where the first pass optimally reads the `filter` columns (i.e. columns that appear in the filter
+ * expression) and the second pass optimally reads the `payload` columns (i.e. columns that do not
+ * appear in the filter expression).
+ */
+class hybrid_scan_reader {
```

Comment:
IMO we should get rid of the `get_` prefixes in this class, unless there's a specific reason to have them.

Reply:
No particular reason, I can remove the prefixes.

Reply:
I think we should. it's just extra letters.

Reply:
All done.